diff --git a/expat/CMakeLists.txt b/expat/CMakeLists.txt
index 0e01f74e..b8f8303e 100644
--- a/expat/CMakeLists.txt
+++ b/expat/CMakeLists.txt
@@ -963,25 +963,6 @@ expat_install(
         expat::
 )
 
-#
-# CPack
-#
-
-# This effectively disables target "package_source".
-# That is done due to CPack's unfortunate choice of an exclusion list
-# rather than inclusion list.  An exclusion list does not protect against
-# unwanted files ending up in the resulting archive in a way that's
-# safe to run from an Expat developer's machine.
-set(CPACK_SOURCE_GENERATOR '')
-
-if(WIN32)
-    set(CPACK_GENERATOR ZIP)
-else()
-    set(CPACK_GENERATOR TGZ)
-endif()
-
-include(CPack)
-
 #
 # Summary
 #
