diff --git a/config/ImathSetup.cmake b/config/ImathSetup.cmake
index ceb73e4..8ed81b0 100644
--- a/config/ImathSetup.cmake
+++ b/config/ImathSetup.cmake
@@ -59,7 +59,7 @@ set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
 # Suffix for debug configuration libraries (if you should choose to
 # install those)
-set(CMAKE_DEBUG_POSTFIX "_d" CACHE STRING "Suffix for debug builds")
+set(CMAKE_DEBUG_POSTFIX "_d")
 
 # Usual cmake option to build shared libraries or not
 option(BUILD_SHARED_LIBS "Build shared library" ON)
