diff --git a/share/cmake/modules/install/Installsse2neon.cmake b/share/cmake/modules/install/Installsse2neon.cmake
index ced4ae13..cd51a6f7 100644
--- a/share/cmake/modules/install/Installsse2neon.cmake
+++ b/share/cmake/modules/install/Installsse2neon.cmake
@@ -13,7 +13,7 @@
 
 include(FetchContent)
 
-set(FETCHCONTENT_BASE_DIR "${CMAKE_BINARY_DIR}/ext/build/sse2neon")
+set(FETCHCONTENT_BASE_DIR "${PROJECT_BINARY_DIR}/ext/build/sse2neon")
 FetchContent_Declare(sse2neon
   GIT_REPOSITORY https://github.com/DLTcollab/sse2neon.git
   GIT_TAG        227cc413fb2d50b2a10073087be96b59d5364aea
@@ -26,7 +26,7 @@ FetchContent_GetProperties(sse2neon)
 if(NOT sse2neon_POPULATED)
   FetchContent_MakeAvailable(sse2neon)
 
-  set(_EXT_DIST_INCLUDE "${CMAKE_BINARY_DIR}/ext/dist/${CMAKE_INSTALL_INCLUDEDIR}")
+  set(_EXT_DIST_INCLUDE "${PROJECT_BINARY_DIR}/ext/dist/${CMAKE_INSTALL_INCLUDEDIR}")
   file(COPY "${sse2neon_SOURCE_DIR}/sse2neon.h" DESTINATION "${_EXT_DIST_INCLUDE}/sse2neon")
 
   # sse2neon_INCLUDE_DIR is used internally for CheckSupportSSEUsingSSE2NEON.cmake and to create sse2neon
diff --git a/share/cmake/utils/CheckSupportAVX.cmake b/share/cmake/utils/CheckSupportAVX.cmake
index fd040b08..eab3aca2 100644
--- a/share/cmake/utils/CheckSupportAVX.cmake
+++ b/share/cmake/utils/CheckSupportAVX.cmake
@@ -36,12 +36,12 @@ set(AVX_CODE "
     }
 ")
 
-file(WRITE "${CMAKE_BINARY_DIR}/CMakeTmp/avx_test.cpp" "${AVX_CODE}")
+file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp/avx_test.cpp" "${AVX_CODE}")
 
 message(STATUS "Performing Test COMPILER_SUPPORTS_AVX")
 try_compile(COMPILER_SUPPORTS_AVX
-  "${CMAKE_BINARY_DIR}/CMakeTmp"
-  "${CMAKE_BINARY_DIR}/CMakeTmp/avx_test.cpp"
+  "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp"
+  "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp/avx_test.cpp"
 )
 
 if(COMPILER_SUPPORTS_AVX)
diff --git a/share/cmake/utils/CheckSupportAVX2.cmake b/share/cmake/utils/CheckSupportAVX2.cmake
index f5a75442..e04b541b 100644
--- a/share/cmake/utils/CheckSupportAVX2.cmake
+++ b/share/cmake/utils/CheckSupportAVX2.cmake
@@ -44,12 +44,12 @@ set(AVX2_CODE "
     }
 ")
 
-file(WRITE "${CMAKE_BINARY_DIR}/CMakeTmp/avx2_test.cpp" "${AVX2_CODE}")
+file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp/avx2_test.cpp" "${AVX2_CODE}")
 
 message(STATUS "Performing Test COMPILER_SUPPORTS_AVX2")
 try_compile(COMPILER_SUPPORTS_AVX2
-  "${CMAKE_BINARY_DIR}/CMakeTmp"
-  "${CMAKE_BINARY_DIR}/CMakeTmp/avx2_test.cpp"
+  "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp"
+  "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp/avx2_test.cpp"
 )
 
 if(COMPILER_SUPPORTS_AVX2)
diff --git a/share/cmake/utils/CheckSupportAVX512.cmake b/share/cmake/utils/CheckSupportAVX512.cmake
index 98004b1d..dbb2fa87 100644
--- a/share/cmake/utils/CheckSupportAVX512.cmake
+++ b/share/cmake/utils/CheckSupportAVX512.cmake
@@ -34,12 +34,12 @@ set(AVX512_CODE "
     }
 ")
 
-file(WRITE "${CMAKE_BINARY_DIR}/CMakeTmp/avx512_test.cpp" "${AVX512_CODE}")
+file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp/avx512_test.cpp" "${AVX512_CODE}")
 
 message(STATUS "Performing Test COMPILER_SUPPORTS_AVX512")
 try_compile(COMPILER_SUPPORTS_AVX512
-  "${CMAKE_BINARY_DIR}/CMakeTmp"
-  "${CMAKE_BINARY_DIR}/CMakeTmp/avx512_test.cpp"
+  "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp"
+  "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp/avx512_test.cpp"
 )
 
 if(COMPILER_SUPPORTS_AVX512)
diff --git a/share/cmake/utils/CheckSupportF16C.cmake b/share/cmake/utils/CheckSupportF16C.cmake
index 45672d67..38b43d27 100644
--- a/share/cmake/utils/CheckSupportF16C.cmake
+++ b/share/cmake/utils/CheckSupportF16C.cmake
@@ -32,12 +32,12 @@ set(F16C_CODE "
     }
 ")
 
-file(WRITE "${CMAKE_BINARY_DIR}/CMakeTmp/f16c_test.cpp" "${F16C_CODE}")
+file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp/f16c_test.cpp" "${F16C_CODE}")
 
 message(STATUS "Performing Test COMPILER_SUPPORTS_F16C")
 try_compile(COMPILER_SUPPORTS_F16C
-  "${CMAKE_BINARY_DIR}/CMakeTmp"
-  "${CMAKE_BINARY_DIR}/CMakeTmp/f16c_test.cpp"
+  "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp"
+  "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp/f16c_test.cpp"
 )
 
 if(COMPILER_SUPPORTS_F16C)
diff --git a/share/cmake/utils/CheckSupportSSE2.cmake b/share/cmake/utils/CheckSupportSSE2.cmake
index 6a07155d..3570503f 100644
--- a/share/cmake/utils/CheckSupportSSE2.cmake
+++ b/share/cmake/utils/CheckSupportSSE2.cmake
@@ -45,12 +45,12 @@ set(SSE2_CODE "
     }
 ")
 
-file(WRITE "${CMAKE_BINARY_DIR}/CMakeTmp/sse2_test.cpp" "${SSE2_CODE}")
+file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp/sse2_test.cpp" "${SSE2_CODE}")
 
 message(STATUS "Performing Test COMPILER_SUPPORTS_SSE2")
 try_compile(COMPILER_SUPPORTS_SSE2
-  "${CMAKE_BINARY_DIR}/CMakeTmp"
-  "${CMAKE_BINARY_DIR}/CMakeTmp/sse2_test.cpp"
+  "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp"
+  "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp/sse2_test.cpp"
 )
 
 if(COMPILER_SUPPORTS_SSE2)
diff --git a/share/cmake/utils/CheckSupportSSE3.cmake b/share/cmake/utils/CheckSupportSSE3.cmake
index ef1dbd31..5e8b6bd6 100644
--- a/share/cmake/utils/CheckSupportSSE3.cmake
+++ b/share/cmake/utils/CheckSupportSSE3.cmake
@@ -32,12 +32,12 @@ set(SSE3_CODE "
     }
 ")
 
-file(WRITE "${CMAKE_BINARY_DIR}/CMakeTmp/sse3_test.cpp" "${SSE3_CODE}")
+file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp/sse3_test.cpp" "${SSE3_CODE}")
 
 message(STATUS "Performing Test COMPILER_SUPPORTS_SSE3")
 try_compile(COMPILER_SUPPORTS_SSE3
-  "${CMAKE_BINARY_DIR}/CMakeTmp"
-  "${CMAKE_BINARY_DIR}/CMakeTmp/sse3_test.cpp"
+  "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp"
+  "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp/sse3_test.cpp"
 )
 
 if(COMPILER_SUPPORTS_SSE3)
diff --git a/share/cmake/utils/CheckSupportSSE4.cmake b/share/cmake/utils/CheckSupportSSE4.cmake
index dd41624d..0d0b425a 100644
--- a/share/cmake/utils/CheckSupportSSE4.cmake
+++ b/share/cmake/utils/CheckSupportSSE4.cmake
@@ -32,12 +32,12 @@ set(SSE4_CODE "
     }
 ")
 
-file(WRITE "${CMAKE_BINARY_DIR}/CMakeTmp/sse4_test.cpp" "${SSE4_CODE}")
+file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp/sse4_test.cpp" "${SSE4_CODE}")
 
 message(STATUS "Performing Test COMPILER_SUPPORTS_SSE4")
 try_compile(COMPILER_SUPPORTS_SSE4
-  "${CMAKE_BINARY_DIR}/CMakeTmp"
-  "${CMAKE_BINARY_DIR}/CMakeTmp/sse4_test.cpp"
+  "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp"
+  "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp/sse4_test.cpp"
 )
 
 if(COMPILER_SUPPORTS_SSE4)
diff --git a/share/cmake/utils/CheckSupportSSE42.cmake b/share/cmake/utils/CheckSupportSSE42.cmake
index 0e0afe23..ee43edb9 100644
--- a/share/cmake/utils/CheckSupportSSE42.cmake
+++ b/share/cmake/utils/CheckSupportSSE42.cmake
@@ -32,12 +32,12 @@ set(SSE42_CODE "
     }
 ")
 
-file(WRITE "${CMAKE_BINARY_DIR}/CMakeTmp/sse42_test.cpp" "${SSE42_CODE}")
+file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp/sse42_test.cpp" "${SSE42_CODE}")
 
 message(STATUS "Performing Test COMPILER_SUPPORTS_SSE42")
 try_compile(COMPILER_SUPPORTS_SSE42
-  "${CMAKE_BINARY_DIR}/CMakeTmp"
-  "${CMAKE_BINARY_DIR}/CMakeTmp/sse42_test.cpp"
+  "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp"
+  "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp/sse42_test.cpp"
 )
 
 if(COMPILER_SUPPORTS_SSE42)
diff --git a/share/cmake/utils/CheckSupportSSEUsingSSE2NEON.cmake b/share/cmake/utils/CheckSupportSSEUsingSSE2NEON.cmake
index d24cda55..c3be92e1 100644
--- a/share/cmake/utils/CheckSupportSSEUsingSSE2NEON.cmake
+++ b/share/cmake/utils/CheckSupportSSEUsingSSE2NEON.cmake
@@ -49,12 +49,12 @@ if((APPLE OR WIN32) AND COMPILER_SUPPORTS_ARM_NEON)
         }
     ")
 
-    file(WRITE "${CMAKE_BINARY_DIR}/CMakeTmp/sse2neon_test.cpp" "${SSE2NEON_CODE}")
+    file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp/sse2neon_test.cpp" "${SSE2NEON_CODE}")
 
     message(STATUS "Performing Test COMPILER_SUPPORTS_SSE_WITH_SSE2NEON")
     try_compile(COMPILER_SUPPORTS_SSE_WITH_SSE2NEON
-      "${CMAKE_BINARY_DIR}/CMakeTmp"
-      "${CMAKE_BINARY_DIR}/CMakeTmp/sse2neon_test.cpp"
+      "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp"
+      "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp/sse2neon_test.cpp"
       COMPILE_DEFINITIONS "-I${sse2neon_INCLUDE_DIR}"
     )
 
diff --git a/share/cmake/utils/CheckSupportSSSE3.cmake b/share/cmake/utils/CheckSupportSSSE3.cmake
index ffa114a7..c02b95db 100644
--- a/share/cmake/utils/CheckSupportSSSE3.cmake
+++ b/share/cmake/utils/CheckSupportSSSE3.cmake
@@ -32,12 +32,12 @@ set(SSSE3_CODE "
     }
 ")
 
-file(WRITE "${CMAKE_BINARY_DIR}/CMakeTmp/ssse3_test.cpp" "${SSSE3_CODE}")
+file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp/ssse3_test.cpp" "${SSSE3_CODE}")
 
 message(STATUS "Performing Test COMPILER_SUPPORTS_SSSE3")
 try_compile(COMPILER_SUPPORTS_SSSE3
-  "${CMAKE_BINARY_DIR}/CMakeTmp"
-  "${CMAKE_BINARY_DIR}/CMakeTmp/ssse3_test.cpp"
+  "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp"
+  "${CMAKE_CURRENT_BINARY_DIR}/CMakeTmp/ssse3_test.cpp"
 )
 
 if(COMPILER_SUPPORTS_SSSE3)
diff --git a/src/OpenColorIO/ops/fixedfunction/FixedFunctionOpCPU.cpp b/src/OpenColorIO/ops/fixedfunction/FixedFunctionOpCPU.cpp
index ed7fcb36..f7ce7ad3 100644
--- a/src/OpenColorIO/ops/fixedfunction/FixedFunctionOpCPU.cpp
+++ b/src/OpenColorIO/ops/fixedfunction/FixedFunctionOpCPU.cpp
@@ -2129,7 +2129,7 @@ __m128 Renderer_PQ_TO_LIN_SSE<true>::myPower(__m128 x, __m128 exp)
     return ssePower(x, exp);
 }
 
-#if (_MSC_VER >= 1920) && (OCIO_USE_AVX)
+#if (_MSC_VER >= 1920) && (OCIO_USE_AVX) && !defined(__clang__)
 // MSVC 2019+ has built-in _mm_pow_ps() SVML intrinsic implementation
 // accessible through immintrin.h. Therefore precise SIMD version is available
 // only when compiled with MSVC and AVX support.
@@ -2184,7 +2184,7 @@ __m128 Renderer_LIN_TO_PQ_SSE<true>::myPower(__m128 x, __m128 exp)
     return ssePower(x, exp);
 }
 
-#if (_MSC_VER >= 1920) && (OCIO_USE_AVX)
+#if (_MSC_VER >= 1920) && (OCIO_USE_AVX) && !defined(__clang__)
 // Only Windows compilers have built-in _mm_pow_ps() SVML intrinsic
 // implementation, so non-fast SIMD version is available only on Windows for
 // now.
@@ -2193,7 +2193,7 @@ __m128 Renderer_LIN_TO_PQ_SSE<false>::myPower(__m128 x, __m128 exp)
 {
     return _mm_pow_ps(x, exp);
 }
-#endif // (_MSC_VER >= 1920) && (OCIO_USE_AVX)
+#endif // (_MSC_VER >= 1920) && (OCIO_USE_AVX) && !defined(__clang__)
 
 template<bool FAST_POWER>
 void Renderer_LIN_TO_PQ_SSE<FAST_POWER>::apply(const void* inImg, void* outImg, long numPixels) const
@@ -2561,7 +2561,7 @@ ConstOpCPURcPtr GetFixedFunctionCPURenderer(ConstFixedFunctionOpDataRcPtr & func
             {
                 return std::make_shared<Renderer_LIN_TO_PQ_SSE<true>>(func);
             }
-#if (_MSC_VER >= 1920) && (OCIO_USE_AVX)
+#if (_MSC_VER >= 1920) && (OCIO_USE_AVX) && !defined(__clang__)
             // MSVC 2019+ has built-in _mm_pow_ps() SVML intrinsic
             // implementation accessible through immintrin.h. Therefore precise
             // SIMD version is available only when compiled with MSVC and AVX
@@ -2578,7 +2578,7 @@ ConstOpCPURcPtr GetFixedFunctionCPURenderer(ConstFixedFunctionOpDataRcPtr & func
             {
                 return std::make_shared<Renderer_PQ_TO_LIN_SSE<true>>(func);
             }
-#if (_MSC_VER >= 1920) && (OCIO_USE_AVX)
+#if (_MSC_VER >= 1920) && (OCIO_USE_AVX) && !defined(__clang__)
             // MSVC 2019+ has built-in _mm_pow_ps() SVML intrinsic
             // implementation accessible through immintrin.h. Therefore precise
             // SIMD version is available only when compiled with MSVC and AVX
